src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1009,7 +1009,7 @@ void MetaspaceShared::initialize_runtime_shared_and_meta_spaces() {
     // The base archive cannot be mapped. We cannot dump the dynamic shared archive.
     AutoCreateSharedArchive = false;
     DynamicDumpSharedSpaces = false;
-    FileMapInfo::fail_continue("Unable to map shared spaces");
+    log_info(cds)("Unable to map shared spaces");
```

Comment:
Need to exit the VM if `RequireSharedSpaces` is true (this replaces the logic in FileMapInfo::fail_continue). Otherwise, "java -Xshare:on" will continue execution even if the archive(s) fails to map.